kvbc/include/categorization/blocks.h

```diff
@@ -0,0 +1,159 @@
+#include "updates.h"
+#include <utility>
+#include <block_digest.h>
+#include "details.h"
+#include "merkle_tree_serialization.h"
+#include "merkle_tree_key_manipulator.h"
+#include "sliver.hpp"
+
+namespace concord::kvbc::categorization {
+
+// Block is composed out of:
+// - BlockID
+// - Parent digest
+// - Categories' updates_info where update info is essentially a list of keys, metadata on the keys and a hash of the
+// category state.
+struct Block {
+  static const std::string CATEGORY_ID;
+
+  Block() = default;
+  Block(const BlockId block_id) { data.block_id = block_id; }
+
+  void add(const std::string& category_id, MerkleUpdatesInfo&& updates_info) {
+    data.categories_updates_info.emplace(std::make_pair(category_id, std::move(updates_info)));
```

Comment:
Nit: I think we can omit `std::make_pair()` as we are doing emplace() and we can just pass the arguments.